projects/plugins/boost/app/modules/optimizations/lcp/class-lcp.php

```diff
@@ -377,6 +378,55 @@ private function optimize_image( $tag ) {
 			$tag = preg_replace( '/<img\s/i', '<img loading="eager" ', $tag );
 		}
 
+		$tag = $this->optimize_image_tag_srcset( $tag );
+
+		return $tag;
+	}
+
+	/**
+	 * Optimize an image tag by adding srcset and sizes attributes.
+	 *
+	 * @param string $tag The original image tag.
+	 * @return string The optimized image tag.
+	 *
+	 * @since $$next-version$$
+	 */
+	private function optimize_image_tag_srcset( $tag ) {
+		if ( preg_match( '/src\s*=\s*["\']([^"\']+)["\']/i', $tag, $matches ) ) {
+			$image_url = $matches[1];
+
+			$image_sizes = array( 300, 600, 900, 1200, 1600 );
+			$sizes       = array();
+			foreach ( $image_sizes as $size ) {
+				$sizes[ $size ] = array( 'w' => $size );
+			}
+
+			// Generate srcset
+			$srcset = array();
+			foreach ( $sizes as $width => $args ) {
+				$srcset[] = Image_CDN_Core::cdn_url( $image_url, $args ) . ' ' . $width . 'w';
+			}
```

Comment:
Why not use one foreach to build both the sizes and the srcset?

```suggestion
			$srcset      = array();
			foreach ( $image_sizes as $width ) {
				$args = array( 'w' => $width );
				$sizes[ $width ] = $args;
				$srcset[] = Image_CDN_Core::cdn_url( $image_url, $args ) . ' ' . $width . 'w';
			}
```

Reply:
On second thought, we don't need to add the args to the sizes. In "// Add sizes attribute if not present" we can just use the `image_sizes` array.